mmpose/datasets/datasets/animal/animal_ap10k_dataset.py

```diff
@@ -282,11 +282,19 @@ def evaluate(self, results, res_folder=None, metric='mAP', **kwargs):
 
         self._write_coco_keypoint_results(valid_kpts, res_file)
 
-        info_str = self._do_python_keypoint_eval(res_file)
-        name_value = OrderedDict(info_str)
+        # do evaluation only if the ground truth keypoint annotations exist
+        if 'annotations' in self.coco.dataset:
+            info_str = self._do_python_keypoint_eval(res_file)
+            name_value = OrderedDict(info_str)
```

Comment:
Otherwise raise exception or warning?

Reply:
For now, it will raise a warning since it has finished this step `_write_coco_keypoint_results`.